spec/controllers/profile_controller_spec.rb

```diff
@@ -1,60 +1,64 @@
 require 'rails_helper'
 
 RSpec.describe ProfileController, type: :controller do
+  let! (:user) { create(:user, username: 'mariasousa', email: 'mariasousa@mail.com') }
 
   describe 'GET #me' do
-    context 'when logged' do
-      login_user
-      it 'returns HTTP status 200' do
-        get :me
-        expect(response).to be_successful
-      end
+    login_user
+    it { should use_before_action(:authenticate_user!) }
+
+    before { get :me }
+    it { is_expected.to respond_with(200) }
+  end
+
+  describe 'GET #show' do
+    login_user
+    it { should use_before_action(:authenticate_user!) }
+
+    context 'If the user it\'s found' do
+      before { post :show, params: { username: user.username } }
+      it { is_expected.to respond_with(200) }
     end
 
-    context 'when not logged in' do
-      it 'redirects to login page' do
-        get :me
-        expect(response).to redirect_to(new_user_session_path)
-      end
+    context 'If the user it\'s not found' do
+      before { post :show, params: { username: 'xyzasas' } }
+      it { is_expected.to respond_with(404) }
+      it { is_expected.to render_template('public/404.html.erb') }
     end
   end
 
-  describe 'get #show' do
-    context 'when logged' do
-      login_user
+  describe 'GET #search' do
+    login_user
+    it { should use_before_action(:authenticate_user!) }
 
-      context 'if the user it\'s found' do
-        before do
-          create(:user, username: 'mariasousa', email: 'mariasousa@mail.com')
-        end
+    before { get :search }
+    it { is_expected.to respond_with(200) }
+    it { is_expected.to render_template('explore') }
 
-        it 'returns HTTP status 200' do
-          get :show, params: { username: 'mariasousa' }
-          expect(response).to have_http_status(200)
-        end
-        it 'renders show template' do
-          get :show, params: { username: 'mariasousa' }
-          expect(response).to render_template('show')
+    context 'When passing a username to search' do
+      context 'If the user it\'s found' do
+        it 'Returns a list of match user(s)' do
```

Comment:
r minúsculo.
Describe começa com maiúsculo ou usa nome do método.
context e it começam com letras minúsculas